app/service/rpc_service.ts

```diff
@@ -164,25 +171,40 @@ class RpcService {
     try {
       response = await fetch(url, { ...init, headers });
     } catch (e) {
-      throw `connection error: ${e}`;
+      throw structuredErrors ? new FetchError(e) : `connection error: ${e}`;
     }
     if (response.status < 200 || response.status >= 400) {
       // Read error message from response body
-      let message = "";
+      let body = "";
       try {
-        message = await response.text();
+        body = await response.text();
       } catch (e) {
-        message = `unknown (failed to read response body: ${e})`;
+        if (structuredErrors) {
+          // If we failed to read the response body then ignore the status code
```

Comment:
Not sure how I feel about this precedence--I guess we still have the network tab to tell the difference, I just figure if a 500 error yielded an unparseable response then saying "500" is more useful than "failed to read".

Reply:
I was thinking that this approach would play more nicely with retry middleware (on the frontend) if we return a generic network error here. That way if we return an error status but fail to read the response body, the client can at least retry for a chance to get a more informative error message that could maybe help them fix the error. I guess if the client got permanently disconnected though, they will be stuck with the "failed to read" error, but I think that's not too much worse than "500: <failed to read error message>"

Reply:
Yeah alright, i'm sold